packages/react-instantsearch-core/src/core/createIndex.js

```diff
@@ -9,14 +9,16 @@ import Index from '../components/Index';
  * @return {object} a Index root
  */
 const createIndex = defaultRoot => {
-  const CreateIndex = ({ indexName, root, children }) => (
-    <Index indexName={indexName} root={root}>
+  const CreateIndex = ({ indexName, indexId, root, children }) => (
+    <Index indexName={indexName} indexId={indexId || indexName} root={root}>
```

Comment:
is there a reason we require (or we'll require later) the user to provide an indexId?
Can't we just generate them from the indexName and make sure they are unique?

Reply:
yes, only on this layer, but it's required in `<Index />` and my point what that if we generate indexId we won't need to expose it and anticipate the case it isn't provided.

Discussed this live with @samouss. It turns out we can't because
- indexIds do not have to be unique 
- there are use cases where user would want to reuse an indexId

Reply:
Yep like @tkrugg said: it can be re-used in some cases. On [the latest PR](https://github.com/algolia/react-instantsearch/pull/1844) I asked about the fact to put it required or not. I'm not sure it brings value to put it required, it brings more constraints to user that won't use it.